File: src/main/java/com/cogent/insurance/service/CustomerPolicyService.java
package com.cogent.insurance.service;

import com.cogent.insurance.shared.dto.CustomerPolicyDto;

import java.util.List;

public interface CustomerPolicyService {

  CustomerPolicyDto createCustomerPolicy(CustomerPolicyDto customerPolicyDto);

  CustomerPolicyDto getCustomerPolicy(String id);

  CustomerPolicyDto updateCustomerPolicy(String id, CustomerPolicyDto customerPolicyDto);

  void deleteCustomerPolicy(String id);

  List<CustomerPolicyDto> getAllCustomerPolicies();
}

File: IMS-angular-client/src/app/service/customer-policy.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class CustomerPolicyService {

  private baseUrl = 'http://localhost:8080/cogent-insurance/api/customer-policies';

  constructor(private http: HttpClient) {
  }

  getCustomerPolicy(customerPolicyId: string): Observable<any> {
    return this.http.get(this.baseUrl + '/' + customerPolicyId);
  }

  createCustomerPolicy(customerPolicy: Object): Observable<Object> {
    return this.http.post(this.baseUrl, customerPolicy);
  }

  updateCustomerPolicy(customerPolicy: Object, id: string): Observable<Object> {
    return this.http.put(this.baseUrl + '/' + id, customerPolicy);
  }

  deleteCustomerPolicy(id: string): Observable<any> {
    return this.http.delete(this.baseUrl + '/' + id);
  }

  getAllCustomerPolicies(): Observable<any> {
    return this.http.get(this.baseUrl);
  }
}
